### PR TITLE
WFLY-14359 ProtoStream marshallers for Object[] should not use packed encoding

### PR DESCRIPTION
**Pull request by pferraro**
Also, WFLY-14270 ProtoStream proxy marshalling should not imply JBoss Modules
Ensures all java.lang.* marshallers use valid protobuf encoding.

https://issues.redhat.com/browse/WFLY-14359
https://issues.redhat.com/browse/WFLY-14270

### CODE REVIEW

**Comment by pferraro** [comment]
@rhusar Please review.

File: clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+/**
+ * {@link ScalarProtoStreamWriter} implementation that does not write to any stream, but instead computes the number of bytes that would be written to a stream.
+ * @author Paul Ferraro
+ */
+public class SizeComputingProtoStreamWriter implements ProtoStreamWriter, Supplier<OptionalInt> {
+
+    private final ImmutableSerializationContext context;
+    private int size = 0;
+    private boolean present = true;
```
Comment (bstansberry) [inline clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java:43]: @pferraro I can't see any place where this is set to any value but 'true'.

File: clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/LoadedClassField.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+package org.wildfly.clustering.marshalling.protostream;
+
+import java.io.IOException;
+import java.io.InvalidClassException;
+
+import org.infinispan.protostream.impl.WireFormat;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * A class field that marshals instances of {@link Class} using a {@link ClassResolver}.
```
Comment (bstansberry) [inline clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/LoadedClassField.java:32]: s/ClassResolver/ClassLoaderMarshaller ?

File: clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java (hunk 1 above)
Comment (pferraro) [reply clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java:43]: https://github.com/pferraro/wildfly/blob/WFLY-14359/clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java#L235

File: clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/LoadedClassField.java (hunk 2 above)
Comment (pferraro) [reply clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/LoadedClassField.java:32]: Oops - just noticed that my IDE's javadoc validation settings were disabled.  Will enable and correct here and elsewhere.

File: clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java (hunk 1 above)
Comment (bstansberry) [reply clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/SizeComputingProtoStreamWriter.java:43]: Thanks

**Review by bstansberry — APPROVED** [review]

**Pull request merged**